modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy

```diff
@@ -354,24 +350,16 @@ class K8sTaskHandler extends TaskHandler implements FusionAwareTask {
             if (s.status == 'FAILED' && s.systemError != null) {
                 task.error = new PodUnschedulableException((String) s.systemError, new Exception("failed to launch pod"))
                 task.aborted = true
-            } else if (s.status == 'FAILED' && s.runtimeError != null) {
-                String err = (String) s.runtimeError
-                if (err.contains('OutOfcpu')) {
-                    throw new K8sOutOfCpuException(err)
-                } else if (err.contains('OutOfmemory')) {
-                    throw new K8sOutOfMemoryException(err)
-                }
-
-                throw new ProcessFailedException(err)
             } else {
                 // finalize the task
                 // note(taras): need to read exitFile first to wait for the files to be available from OFS
                 task.exitStatus = readExitFile()
                 if (task.exitStatus == Integer.MAX_VALUE && s.exitCode != null) {
                     task.exitStatus = ((String) s.exitCode).toInteger()
                 }
+
                 task.stdout = outputFile
-                task.stderr = errorFile
+                task.stderr = s.runtimeError != null ? (String) s.runtimeError : errorFile
```

Comment:
```suggestion
                task.stderr = (String) (s.runtimeError ?: errorFile)
```

Reply:
error File is a Path object not a string